drivers/sensor/stm32_temp/stm32_temp.c

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) 2021 Eug Krashtan
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <device.h>
+#include <drivers/sensor.h>
+#include <drivers/adc.h>
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(stm32_temp, CONFIG_SENSOR_LOG_LEVEL);
+
+#define DT_DRV_COMPAT st_stm32_temp
+
+#define STM32_TEMP DT_INST(0, st_stm32_temp)
+
+/*
+ * Average slope of T-V chart according to datasheet
+ * "Electrical characteristics/Operating conditions"
+ */
+#if defined(CONFIG_SOC_SERIES_STM32F1X)
+/*
+ * Table 5.3.19 (in mV/C)
+ * (min is 4 mV/C, max 4.6, default (4.3): typical)
+ */
+#define AVGSLOPE 4.3
+#else /* CONFIG_SOC_SERIES_STM32F1X */
+/*
+ * Table 6.3.21 (in mV/C)
+ */
+#define AVGSLOPE 2.5
+#endif /* CONFIG_SOC_SERIES_STM32F1X */
+
+/*
+ * Voltage of temperature sensor at 25C according to datasheet
+ * "Electrical characteristics"
+ */
+#if defined(CONFIG_SOC_SERIES_STM32F1X)
+/*
+ * table 5.3.19 (in V)
+ * (min is 1.34, max is 1.52, default(1.43): typical)
+ */
+#define V25 1.43
+#else /* CONFIG_SOC_SERIES_STM32F1X */
+/*
+ * Table 6.3.21 (in V)
+ */
+#define V25 0.76
+#endif /* CONFIG_SOC_SERIES_STM32F1X */
```

Comment:
If V25 and AVGSLOPE are likely to change between each soc, can we consider getting those from device tree ?

Reply:
@erwango Exactly!

Reply:
@Delsian I checked the driver file. For me it looks that the temperature enabling is not activated  (see https://github.com/zephyrproject-rtos/zephyr/blob/4c2fe4b39c0a29a535fd0fc21a706a53e53a2975/drivers/adc/adc_stm32.c#L697)

What I was mainly referring was the use of the `__LL_ADC_DECIMAL_NB_TO_CHANNEL(index)` in combination with the dts binding.

form stm32F4 HAL
```
_LL_ADC_CHANNEL_TO_DECIMAL_NB
Description:
•
Helper macro to get ADC channel number in decimal format from literals LL_ADC_CHANNEL_x.
Parameters:
•
__CHANNEL__: This parameter can be one of the following values:
–
LL_ADC_CHANNEL_0
–
LL_ADC_CHANNEL_1
–
LL_ADC_CHANNEL_2
–
LL_ADC_CHANNEL_3
–
LL_ADC_CHANNEL_4
–
LL_ADC_CHANNEL_5
–
LL_ADC_CHANNEL_6
–
LL_ADC_CHANNEL_7
–
LL_ADC_CHANNEL_8
–
LL_ADC_CHANNEL_9
–
LL_ADC_CHANNEL_10
–
LL_ADC_CHANNEL_11
–
LL_ADC_CHANNEL_12
–
LL_ADC_CHANNEL_13
–
LL_ADC_CHANNEL_14
–
LL_ADC_CHANNEL_15
–
LL_ADC_CHANNEL_16
–
LL_ADC_CHANNEL_17
–
LL_ADC_CHANNEL_18
–
LL_ADC_CHANNEL_VREFINT (1)
–
LL_ADC_CHANNEL_TEMPSENSOR (1)(2)
–
LL_ADC_CHANNEL_VBAT (1)
Return value:
•
Value: between Min_Data=0 and Max_Data=18
Notes:
•
Example: __LL_ADC_CHANNEL_TO_DECIMAL_NB(LL_ADC_CHANNEL_4) will return decimal number
"4". The input can be a value from functions where a channel number is returned, either defined with
number or with bitfield (only one bit must be set).
```
As you can see at the driver

https://github.com/zephyrproject-rtos/zephyr/blob/6857eb198eb937b8d30c17203e5c46a54d67537b/drivers/adc/adc_stm32.c#L527

this line excludes the VREF, TEMPSENSOR and VBAT

check the define (STM32_CHANNEL_COUNT is set to 20)

I hope that you/we may alter the driver so we can use VREF etc. (What is my goal)

Reply:
Ok, I moved all parameters to DTS overlay, and now both existing boards (F103 and F412) work fine.
Can somebody check with other series and propose overlays with correct values?

Reply:
> @Delsian I checked the driver file. For me it looks that the temperature enabling is not activated for the stm32F4 series (see
> 
> https://github.com/zephyrproject-rtos/zephyr/blob/4c2fe4b39c0a29a535fd0fc21a706a53e53a2975/drivers/adc/adc_stm32.c#L697
> 
> )

@StefJar I suppose we can drop this line as it not used anymore. Now TSVREFE init occurs in channel initialization.